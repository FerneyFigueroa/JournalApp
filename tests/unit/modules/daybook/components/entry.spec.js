import { shallowMount } from "@vue/test-utils";

import EntryMi from '../../../../../src/modules/daybook/components/EntryMi.vue'




import {journalState} from '../../../../../tests/unit/mock-date/test-journal-state'

describe('pruebas en el entry componet',()=>{

    const mockRouter = {
        push: jest.fn()
    }
    const wrapper = shallowMount(EntryMi,{
        props:{
            entry: journalState.entries[0]
        },
        global:{
            mocks:{
                $router: mockRouter
            }
        }
    })

    test('debe hacer match con el snapshot',()=>{
        expect(wrapper.html()).toMatchSnapshot();

    })

    test('debe de redireccionar al hacer click en el entry.container',()=>{

        const entrycontainer =  wrapper.find('.entry-container')

        entrycontainer.trigger('click')

        expect(mockRouter.push).toHaveBeenCalledWith({ 
            name: "entry",
              params: {
               id: journalState.entries[0].id
            }
         })
       


        
    })

    test('pruebas en las propiedades computadas',()=>{
        

        expect(wrapper.vm.day).toBe(26)
        expect(wrapper.vm.month).toBe('Noviembre')
        expect(wrapper.vm.yearDay).toBe('2023, Domingo')
    })


})

