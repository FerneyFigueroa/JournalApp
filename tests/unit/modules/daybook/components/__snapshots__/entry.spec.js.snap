// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el entry componet debe hacer match con el snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">26</span><span class="mx-1 fs-5">Noviembre</span><span class="mx-2 fw-ligth">2023, Domingo</span></div>
  <div class="entry-description">hola esta es la prueba</div>
</div>
`;
